### DISCUSSION TITLE
how to change the dimension of sidesets？

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#20080] [Q&A Meshing] [answered]
hello，
   By the way I tried to get sidesets, I got a 3D network of sidesets index. I noticed that it is possible to use lowerLowerDBlockFromSidesetGenerator to convert 3D sidesets to 2D blocks, but can I turn this 2D block into 2D sidesets?Because I need to calculate the surface heat flux density, I need a 2D boundary. Looking forward to your answer.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I have not done or tested this before, but you should be able to create a 2D sideset from 2D block, at least for the sideset generators where this makes sense. If not, then it's a feature we should make sure to support. 
Just use whichever sideset generator you are wanting to use with the 2D block as the argument. 

Please let us know how this goes,
Guillaume

**Reply by amosaha** [reply]
Yes, only 2D blocks can be used to create 2D sidesets. At present, I have obtained such a 2D block through LowerDBlockFromSidesetGenerator. I would like to know if I can use this 2D plane block to turn it into a sideset. Searched almost all meshgenerator resources, I didn't find a suitable way.

**Reply by GiudGiud** [reply]
Ok just to be absolutely sure, it is '2D sideset' as in a 'sideset of a 2D block', so really a sideset made of 1D elements.

Can you use this one for example?
https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html
With the block argument set to the 2D block you extracted ?

**Comment by amosaha** [comment]
I have tried using this type on my 2D block, and it resulted in the bounding of the 2d face. So if you use a rectangle block, the resulting sidesets are the four sides of the rectangle.









发自我的iPhone


------------------ Original ------------------
From: Guillaume Giudicelli ***@***.***&gt;
Date: Wed,Jan 19,2022 11:03 PM
To: idaholab/moose ***@***.***&gt;
Cc: amosaha ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [idaholab/moose] how to change the dimension of sidesets？ (Discussion #20080)





 
Ok just to be absolutely sure, it is '2D sideset' as in a 'sideset of a 2D block', so really a sideset made of 1D elements.
 
Can you use this one for example?
 https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html
 With the block argument set to the 2D block you extracted ?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GiudGiud** [reply]
Isnt that what you want?

**Comment by amosaha** [comment]
No, I want the sidesets of the whole plane.




发自我的iPhone


------------------ Original ------------------
From: Guillaume Giudicelli ***@***.***&gt;
Date: Wed,Jan 19,2022 11:36 PM
To: idaholab/moose ***@***.***&gt;
Cc: amosaha ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [idaholab/moose] how to change the dimension of sidesets？ (Discussion #20080)





 
Isnt that what you want?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GiudGiud** [reply]
so you want a 2D surface?
Or you want all the 1D elements in the 2D block?

**Comment by amosaha** [comment]
I want a 2d surface as the sidesets.



发自我的iPhone


------------------ Original ------------------
From: Guillaume Giudicelli ***@***.***&gt;
Date: Thu,Jan 20,2022 8:34 AM
To: idaholab/moose ***@***.***&gt;
Cc: amosaha ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [idaholab/moose] how to change the dimension of sidesets？ (Discussion #20080)





 
so you want a 2D surface?
 Or you want all the 1D elements in the 2D block?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GiudGiud** [reply]
the use the sideset generators on the 3D mesh directly, not the lowerDblock...generator

**Comment by amosaha** [comment]
But the sidesets I get directly using 3D block are not really 2D faces, just an index of a 3D block.

发自我的iPhone


------------------ Original ------------------
From: Guillaume Giudicelli ***@***.***&gt;
Date: Thu,Jan 20,2022 9:00 AM
To: idaholab/moose ***@***.***&gt;
Cc: amosaha ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [idaholab/moose] how to change the dimension of sidesets？ (Discussion #20080)





 
the use the sideset generators on the 3D mesh directly, not the lowerDblock...generator
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GiudGiud** [reply]
not sure what you mean by an index of a 3D block.
They are a collection of a 2D elements, which are faces of 3D elements.
